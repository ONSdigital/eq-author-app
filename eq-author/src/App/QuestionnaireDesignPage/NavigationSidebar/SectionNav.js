import React, { Component } from "react";
import styled from "styled-components";
import CustomPropTypes from "custom-prop-types";
import PropTypes from "prop-types";

import { TransitionGroup } from "react-transition-group";
import NavItemTransition from "./NavItemTransition";
import SectionNavItem from "./SectionNavItem";
import scrollIntoView from "utils/scrollIntoView";
import gql from "graphql-tag";

const NavList = styled.ol`
  margin: 0 0 1em;
  padding: 0;
  list-style: none;
  font-weight: bold;
`;

class SectionNav extends Component {
  static propTypes = {
    questionnaire: CustomPropTypes.questionnaire,
    isOpen: PropTypes.shape({ open: PropTypes.bool }).isRequired,
    handleChange: PropTypes.func,
  };

  render() {
    const { questionnaire, isOpen, handleChange } = this.props;

    return (
      <TransitionGroup component={NavList}>
        {questionnaire.sections.map((section, index) => (
          <NavItemTransition key={section.id} onEntered={scrollIntoView}>
            <SectionNavItem
              questionnaire={questionnaire}
              isOpen={isOpen}
              section={section}
              handleChange={handleChange}
              identity={index}
            />
          </NavItemTransition>
        ))}
      </TransitionGroup>
    );
  }
}

SectionNav.fragments = {
  SectionNav: gql`
    fragment SectionNav on Questionnaire {
      id
      sections {
        ...SectionNavItem
      }
    }
    ${SectionNavItem.fragments.SectionNavItem}
  `,
};

export default SectionNav;
