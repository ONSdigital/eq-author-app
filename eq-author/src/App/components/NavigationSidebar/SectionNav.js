import React, { Component } from "react";
import CustomPropTypes from "custom-prop-types";
import styled from "styled-components";

import { TransitionGroup } from "react-transition-group";
import NavItemTransition from "App/components/NavigationSidebar/NavItemTransition";
import SectionNavItem from "App/components/NavigationSidebar/SectionNavItem";
import scrollIntoView from "utils/scrollIntoView";
import gql from "graphql-tag";

const NavList = styled.ol`
  margin: 0 0 1em;
  padding: 0;
  list-style: none;
  font-weight: bold;
`;

class SectionNav extends Component {
  static propTypes = {
    questionnaire: CustomPropTypes.questionnaire
  };

  render() {
    const { questionnaire } = this.props;
    return (
      <TransitionGroup component={NavList}>
        {questionnaire.sections.map(section => (
          <NavItemTransition key={section.id} onEntered={scrollIntoView}>
            <SectionNavItem questionnaire={questionnaire} section={section} />
          </NavItemTransition>
        ))}
      </TransitionGroup>
    );
  }
}

SectionNav.fragments = {
  SectionNav: gql`
    fragment SectionNav on Questionnaire {
      id
      sections {
        ...SectionNavItem
      }
    }
    ${SectionNavItem.fragments.SectionNavItem}
  `
};

export default SectionNav;
