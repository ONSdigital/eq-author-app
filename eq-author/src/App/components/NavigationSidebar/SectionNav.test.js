import React from "react";
import { shallow } from "enzyme";
import SectionNav from "App/components/NavigationSidebar/SectionNav";

describe("SectionNav", () => {
  let wrapper;

  const page = { id: "2", title: "Page" };
  const section = { id: "3", title: "Section", pages: [page] };
  const questionnaire = {
    id: "1",
    title: "Questionnaire",
    sections: [section]
  };

  beforeEach(() => {
    wrapper = shallow(
      <SectionNav
        questionnaire={questionnaire}
        currentSectionId={section.id}
        currentPageId={page.id}
      />
    );
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
