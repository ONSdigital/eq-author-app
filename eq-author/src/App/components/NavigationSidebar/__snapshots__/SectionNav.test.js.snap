// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionNav should render 1`] = `
<TransitionGroup
  childFactory={[Function]}
  component={[Function]}
>
  <NavItemTransition
    in={false}
    key="3"
    onEntered={[Function]}
    timeout={300}
  >
    <withRouter(UnwrappedSectionNavItem)
      questionnaire={
        Object {
          "id": "1",
          "sections": Array [
            Object {
              "id": "3",
              "pages": Array [
                Object {
                  "id": "2",
                  "title": "Page",
                },
              ],
              "title": "Section",
            },
          ],
          "title": "Questionnaire",
        }
      }
      section={
        Object {
          "id": "3",
          "pages": Array [
            Object {
              "id": "2",
              "title": "Page",
            },
          ],
          "title": "Section",
        }
      }
    />
  </NavItemTransition>
</TransitionGroup>
`;
